olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationCredentials.h

```diff
@@ -59,6 +60,39 @@ class AUTHENTICATION_API AuthenticationCredentials {
    */
   const std::string& GetSecret() const;
 
+  /*
+   * @brief Reads credentials to access OLP from an input stream that is
+   * interpreted as a sequence of characters.
+   *
+   * The stream must contain the following sequences of characters:
+   * here.access.key.id = <user's key id>
+   * here.access.key.secret = <user's key secret>
+   *
+   * @param[in] stream Stream to read the credentials from.
+   * @return Optional value with the credentials if the credentials were read
+   * successfully, or no value in case of failure.
+   */
+  static boost::optional<AuthenticationCredentials> ReadFromStream(
+      std::istream& stream);
+
+  /*
+   * @brief Reads credentials to access OLP from a text file.
+   *
+   * The file must contain the following lines:
+   * here.access.key.id = <user's key id>
+   * here.access.key.secret = <user's key secret>
+   * Such a file can be downloaded from OLP when a new application is
+   * registered.
+   *
+   * @param[in] filename Path to the file that contains the credentials. Empty
+   * path will be replaced with the default path which is
+   * $HOME/.here/credentials.properties
+   * @return Optional value with the credentials if the credentials were read
```

Comment:
Please check the latest revision (e3a90c4) that was pushed 2 hours ago, instead of penultimate.
You will find that this comment is addressed already.